lib/smart_properties/errors.rb

```diff
@@ -35,20 +35,32 @@ class InvalidValueError < AssignmentError
 
     def initialize(sender, property, value)
       @value = value
+
       super(
         sender,
         property,
-        "%s does not accept %s as value for the property %s" % [
+        "%s does not accept %s as value for the property %s. Only accepts: %s" % [
           sender.class.name,
           value.inspect,
-          property.name
+          property.name,
+          acceptor_message(sender, property)
         ]
       )
     end
 
     def to_hash
       Hash[property.name, "does not accept %s as value" % value.inspect]
     end
+
+    private
+
+    def acceptor_message(sender, property)
+      acceptor = sender.class.properties[property.name].accepter
```

Comment:
Seeing this just now after I merged, but couldn't you just use `property.accepter` here?